pkg/kubelet/nodeshutdown/nodeshutdown_manager_linux.go

```diff
@@ -76,14 +79,31 @@ type Manager struct {
 }
 
 // NewManager returns a new node shutdown manager.
-func NewManager(getPodsFunc eviction.ActivePodsFunc, killPodFunc eviction.KillPodFunc, shutdownGracePeriodRequested, shutdownGracePeriodCriticalPods time.Duration) *Manager {
-	return &Manager{
+func NewManager(getPodsFunc eviction.ActivePodsFunc, killPodFunc eviction.KillPodFunc, syncNodeStatus func(), shutdownGracePeriodRequested, shutdownGracePeriodCriticalPods time.Duration) (*Manager, lifecycle.PodAdmitHandler) {
+	manager := &Manager{
 		getPods:                         getPodsFunc,
 		killPod:                         killPodFunc,
+		syncNodeStatus:                  syncNodeStatus,
 		shutdownGracePeriodRequested:    shutdownGracePeriodRequested,
 		shutdownGracePeriodCriticalPods: shutdownGracePeriodCriticalPods,
 		clock:                           clock.RealClock{},
 	}
+	return manager, manager
+}
+
+// Admit rejects all pods if node is shutting
+func (m *Manager) Admit(attrs *lifecycle.PodAdmitAttributes) lifecycle.PodAdmitResult {
+	m.nodeShuttingDownMutex.Lock()
```

Comment:
instead of reading `nodeShuttingDownNow` directly here and dealing with the mutex, I would instead use `ShutdownStatus()` function which already takes care of locking mutex and returning the correct result. It also has a short circuit check if the the feature gate is disabled.

Something like:

```
nodeShuttingDown := m.ShutdownStatus() != nil

if nodeShuttingDown  {
    return lifecycle.PodAdmitResult{
		  Admit:   false,
		  Reason:  nodeShutdownReason,
		  Message: nodeShutdownNotAdmitMessage,
	}
}
return lifecycle.PodAdmitResult{Admit: true}
```